tests/snippets/bools.py

```diff
@@ -39,3 +39,10 @@ def __bool__(self):
 assert (True and True)
 assert not (False and fake)
 assert (True and 5) == 5
+
+# Bools are also ints.
+assert isinstance(True, int)
+assert True + True == 2
```

Comment:
Wait, what? :stuck_out_tongue:

Reply:
Yeah, that was my reaction when I figured this out.

I knew 1 and True were sort of the same, I didn't realise how the same.

Reply:
This is surely insane. We should ask ourselves how compatible do we want rustpython to be with cpython. I mean, here is makes sense to be incompatible I guess?

Reply:
The history is interesting here: https://www.python.org/dev/peps/pep-0285/

This in particular is considered somewhat idiomatic, so we would be breaking existing code.

```
def count_gt10(y):
  return sum(x > 10 for x in y)
```

I guess it depends on the goal of the project. As a learning tool, I enjoy exploring all these weird edge cases of python and this gives us a live example to check inheriting from int works. As a practical tool, if we want to import and run existing python code, we'll need to be compatible here.

I'm kind of surprised this wasn't cleaned up in python 3, but the PEP makes it seem like this wasn't considered a particularly problematic bit of python history.

Reply:
We will see in python 4 :)